client/my-sites/plugins/controller-logged-out.js

```diff
@@ -0,0 +1,73 @@
+import { BASE_STALE_TIME } from 'calypso/data/marketplace/constants';
+import {
+	getFetchWPCOMFeaturedPlugins,
+	getFetchWPCOMPlugins,
+} from 'calypso/data/marketplace/use-wpcom-plugins-query';
+import { getFetchWPORGPlugins } from 'calypso/data/marketplace/use-wporg-plugin-query';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+
+function getProps( context ) {
+	const props = {
+		path: context.path,
+		locale: context.lang,
+		tag: '',
+	};
+	return props;
+}
+
+function prefetchPluginsData( queryClient, fetchParams, infinite ) {
+	const queryType = infinite ? 'prefetchInfiniteQuery' : 'prefetchQuery';
+
+	return queryClient[ queryType ]( ...fetchParams, {
+		staleTime: BASE_STALE_TIME,
+	} );
+}
+
+const prefetchPaidPlugins = ( queryClient, options ) =>
+	prefetchPluginsData(
+		queryClient,
+		getFetchWPCOMPlugins( true, 'all', options.search, options.tag )
```

Comment:
The `getFetchWPCOMPlugins` function returns _parameters_ for the query (key and fn), but that's not communicated by the function name at all. I would call it something like `getWPCOMPluginsQueryParams` instead.

Also, the first parameter (`enabled`) can be removed now when `enabled` is no longer part of the query key.

Reply:
Agreed and renamed them the functions.